keyring/backends/libsecret.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from ..util import properties
+from ..backend import KeyringBackend
+from ..credentials import SimpleCredential
+from ..errors import (
+    PasswordDeleteError,
+    PasswordSetError,
+    ExceptionRaisedContext,
+    KeyringLocked,
+)
+
+available = False
+try:
+    import gi
+    from gi.repository import Gio
+    from gi.repository import GLib
+    gi.require_version('Secret', '1')
+    from gi.repository import Secret
+    available = True
+except (AttributeError, ImportError, ValueError):
+    pass
+
+log = logging.getLogger(__name__)
+
+
+class Keyring(KeyringBackend):
+    """libsecret Keyring"""
+
+    appid = 'Python keyring library'
+    if available:
+        schema = Secret.Schema.new(
+            "org.freedesktop.Secret.Generic",
+            Secret.SchemaFlags.NONE,
+            {
+                "application": Secret.SchemaAttributeType.STRING,
+                "service": Secret.SchemaAttributeType.STRING,
+                "username": Secret.SchemaAttributeType.STRING,
+            }
+        )
+
+    @properties.ClassProperty
+    @classmethod
+    def priority(cls):
+        with ExceptionRaisedContext() as exc:
+            Secret.__name__
+        if exc:
+            raise RuntimeError("libsecret required")
+        return 5
+
+    def get_password(self, service, username):
+        """Get password of the username for the service"""
+        attributes = {
+            "application": self.appid,
+            "service": service,
+            "username": username,
+        }
+        try:
+            items = Secret.password_search_sync(self.schema, attributes,
+                                                Secret.SearchFlags.UNLOCK, None)
+        except GLib.Error as error:
+            quark = GLib.quark_try_string('g-io-error-quark')
+            if error.matches(quark, Gio.IOErrorEnum.FAILED):
+                raise KeyringLocked('Failed to unlock the item!')
```

Comment:
Here and below: maybe add `from error`?
```suggestion
                raise KeyringLocked('Failed to unlock the item!') from error
```